src/language-markdown/utils.js

```diff
@@ -36,38 +36,39 @@ const INLINE_NODE_WRAPPER_TYPES = [
 const kRegex = new RegExp(kPattern);
 const punctuationRegex = new RegExp(punctuationPattern);
 
+const KIND_NON_CJK = "non-cjk";
+const KIND_CJ_LETTER = "cj-letter";
+const KIND_K_LETTER = "k-letter";
+const KIND_CJK_PUNCTUATION = "cjk-punctuation";
+
+/**
+ * @typedef {" " | "\n" | ""} WhiteSpaceValue
+ * @typedef { KIND_NON_CJK | KIND_CJ_LETTER | KIND_K_LETTER | KIND_CJK_PUNCTUATION } WordKind
+ * @typedef {{
+ *   type: "whitespace",
+ *   value: WhiteSpaceValue,
+ *   kind?: undefined,
+ *   hasLeadingPunctuation?: undefined,
+ *   hasTrailingPunctuation?: undefined,
+ * } | {
+ *   type: "word",
+ *   value: string,
+ *   kind: WordKind,
+ *   hasLeadingPunctuation: boolean,
+ *   hasTrailingPunctuation: boolean,
+ * }} TextNode
+ */
+
 /**
  * split text into whitespaces and words
  * @param {string} text
  */
-function splitText(text, options) {
-  const KIND_NON_CJK = "non-cjk";
-  const KIND_CJ_LETTER = "cj-letter";
-  const KIND_K_LETTER = "k-letter";
-  const KIND_CJK_PUNCTUATION = "cjk-punctuation";
-
-  /**
-   * @type {Array<
-   *   { type: "whitespace", value: " " | "\n" | "" }
-   *   | {
-   *     type: "word",
-   *     value: string,
-   *     kind: string,
-   *     hasLeadingPunctuation: boolean,
-   *     hasTrailingPunctuation: boolean
-   *   }
-   * >}
-   */
+// eslint-disable-next-line no-unused-vars
+function splitText(text, _options) {
```

Comment:
No need to keep the `options` parameter.

Reply:
Should we tell ` \n` surrounded by CHinese/Japanese from common `\n`? I am planning to revive the `option` (`.proseWrap`).

---

```
日本語 
日本語
```

↓ `proseWrap = never`

```
日本語 日本語
```

---

```
日本語
日本語
```

↓ `proseWrap = never`

```
日本語日本語
```

Reply:
> I am planning to revive the options (.proseWrap).

What for? Do you want to do that only to support that case with the space character at the end of the line (`<CJ><space><\n><CJ>`) or do you have some other cases in mind?

> Should we tell \n surrounded by Chinese/Japanese from common \n when proseWrap is other than preserve?

Do I understand your question correctly that you're asking here which behavior is better in that case with `<CJ><space><\n><CJ>`? If so, I don't know neither Chinese nor Japanese, so I can't answer this question.

Reply:
I checked the [CommonMark playground](https://spec.commonmark.org/dingus/?text=%E6%97%A5%E6%9C%AC%E8%AA%9E%0A%E6%97%A5%E6%9C%AC%E8%AA%9E%0A%0A%E6%97%A5%E6%9C%AC%E8%AA%9E%20%0A%E6%97%A5%E6%9C%AC%E8%AA%9E). This space doesn't affect the AST and the output, so it seems totally okay to ignore it.

Reply:
But let's add a test for this case.

Reply:
@thorn0 
So do you mean I should do these?

- Remove `options`
- Add a test case for trailing space

Reply:
Depends on your answers to my questions.

Reply:
@thorn0 
https://prettier.io/playground/#N4Igxg9gdgLgprEAuEghBnegBAHSh9u+qOe+IANCBAA4wCW0AzsqAIYBO7EA7gAocJmKVgBturAJ7NKAI3aswAazgwAyqwC2cADJ0ocZADNRjOLPlKVq6gr0BzZDHYBXMyDgaZcACbef21ig7Z1Y7OAAxCHYNVhh6IOQQVmcYCAoQAAsYDREAdQy6eEYbMDhVQUK6ADdCiUSwRmkQPVN2GF55OxijEzcAK0YAD1V7ETgARWcIeB6RU0obdlbEmPZFbx4odOp2PRhcum8YDOQARlOFrlNc+WpE0XEpdIBHKfgOmiEkxgBafR8fOl2HBXnRgR1Qt0kMY5m5TBo6I4XHDRhM3gZob1KDBWDIDkcTkgAEzY+R0ET2ADCEA0UPcjAArOlnKYACq4oQw+YgKquACSUD8sFUYF2tAAgoLVDAJGNZqYAL4KoA

I found a trailing single space will be removed unless unwrapped.
I will do:

> @thorn0 So do you mean I should do these?
> 
>     * Remove `options`
> 
>     * Add a test case for trailing space

Reply:
I found the current .editorconfig does not allow trailling single or double whitespaces. Should I fixit?

Reply:
Use  `snippets` https://github.com/prettier/prettier/blob/bfb09bf214fcdb0e47ed7cb1c702281396c61753/tests/format/html/whitespace/jsfmt.spec.js#L6

Reply:
@thorn0 @fisker 
Updated `tests/format/markdown/splitCjkText/jsfmt.spec.js`